### PR TITLE
build: Add melos command bootstrap dependencies

### PR DESCRIPTION
**Pull request by riscait**
## 🙌 What's Done

- Add melos command bootstrap dependencies
- Add build_runner diff
- Add pubspec.lock files

## ✍️ What's Not Done

-

### CODE REVIEW

File: melos.yaml [hunk 1]
```diff
@@ -16,6 +24,7 @@ command:
         melos exec --flutter --file-exists="macos/Podfile.lock" -- "cd macos && rm Podfile.lock"
   version:
     branch: main
+    releaseUrl: true
```
Comment (riscait) [inline melos.yaml:27]: > Generate and print a link to the prefilled release creation page for each package after versioning. Defaults to false.

https://melos.invertase.dev/commands/version#--no-release-url--r

File: packages/amazon_paapi/lib/src/paapi/get_items/paapi_get_items_parameters.freezed.dart
Comment (riscait) [inline packages/amazon_paapi/lib/src/paapi/get_items/paapi_get_items_parameters.freezed.dart:1]: Difference due to build_runner update

File: packages/rest_api_client/.gitignore
Comment (riscait) [inline packages/rest_api_client/.gitignore:1]: For packages used in monorepo (and which we are utilizing now), we set `pubspec.lock` as the target of git management!

**Review by naipaka — APPROVED** [review]
LGTM
The melos command bootstrap dependencies is very convenient.✍️

**Review by k-nkmr — APPROVED** [review]
ありがとうございます！

**Pull request merged**